---
layout: post
title: New insights into microbial ecology through subtle nucleotide variation
excerpt: "A summary of the oligotyping special topic in Frontiers In Microbiology"
date: 2015-02-12 10:17:05
tags: [pubs, frontiers]
categories: [oligotyping]
comments: true
image:
  feature: header.jpg
  thumb: header.jpg
---

{% include _toc.html %}

>I am very pleased to announce that Frontiers in Microbiology is now hosting a research topic on oligotyping, which is open for submissions! 

This is what [I had said]({% post_url oligotyping/2013-12-12-oligotyping-frontiers %}) on these pages about a year ago. Today, the research topic [New insights into microbial ecology through subtle nucleotide variation](http://journal.frontiersin.org/ResearchTopic/2427) is almost concluded, and contains 8 publications.

I thought this would be a good time to offer a glimpse of what has been published in this collection so far.


## Gut microbiomes of cheetahs and jackals

<figure>
	<a href="http://journal.frontiersin.org/Journal/10.3389/fmicb.2014.00526/full"><img src="{{ site.url }}/images/oligotyping/menke_et_al.png"></a>
</figure>

[In their study](http://journal.frontiersin.org/Journal/10.3389/fmicb.2014.00526/full) Menke *et al*. compare the gut microbiomes of two sympatric mammalian carnivores, cheetah and black-backed jackal, by sequencing the V4 region of the 16S rRNA gene they amplified from fecal material collected from **free-ranging** animals!

Being sympatric, and having somewhat similar diets, these animals seem to have similar gut microbiomes in the big picture, yet their microbiomes are different enough in lower levels of taxonomy for them to separate from each other on an ordination. However, authors show that even genera that seem to be shared among the two are composed of different oligotypes. For instance *Blautia* is pretty abundant in both cheetah and jackal group, yet *Blautia* oligotypes differ dramatically between the two (cheetah samples are on the left) (Figure 6 from Menke *et al*.):

<figure>
	<a href="{{ site.url }}/images/oligotyping/menke_et_al_blautia.png"><img src="{{ site.url }}/images/oligotyping/menke_et_al_blautia.png"></a>
</figure>

The figure shows that *Blautia* group is more abundant in cheetah samples in general, which is how it contributes to the separation of cheetah samples from jackal samples in an ordination. But with oligotypes, beyond separation, there are specific markers for each group from organisms that seem to occur in guts of more species than the other. *Blautia* is especially interesting to me because this group seems to be a great marker for different host species, as we had shown in our [2014 ISMEJ publication](http://www.nature.com/ismej/journal/v9/n1/full/ismej201497a.html). But *Blautia* is not the only genus the oligotype distribution differ between the groups drastically. Here, *Slackia*:

<figure>
	<a href="{{ site.url }}/images/oligotyping/menke_et_al_slackia.png"><img src="{{ site.url }}/images/oligotyping/menke_et_al_slackia.png"></a>
</figure>

You can read the study here: [doi:10.3389/fmicb.2014.00526](http://journal.frontiersin.org/Journal/10.3389/fmicb.2014.00526/full)


## *Arcobacter* in sewage

<figure>
	<a href="http://journal.frontiersin.org/Journal/10.3389/fmicb.2014.00525/full"><img src="{{ site.url }}/images/oligotyping/fisher_et_al.png"></a>
</figure>

To better understand the ecological factors that affect the survival and growth of *Arcobacter* spp. in sewer infrastructure, [Fisher *et al*. dissect the *Arcobacter* group](http://journal.frontiersin.org/Journal/10.3389/fmicb.2014.00525/full) (which comprise 5% to 11% of sewage bacterial  community) into a more precisely-defined taxonomic units by oligotyping reads they amplified from the V4V5 region of the gene.

The study contains sequencing data from 12 sewage treatment centers in the US that were sampled three times: August 2012, January 2013, and April 2013 (there is also one additional sample from Spain). The stability of sewage systems is just very impressive. The first figure in the study shows that although the composition of *Arcobacter* oligotypes differ from one station to another, they don't differ much across the sampling periods within one station (even when their abuncance in the overall sample change quite drastically):

<figure>
	<a href="{{ site.url }}/images/oligotyping/fisher_et_al_fig_1.png"><img src="{{ site.url }}/images/oligotyping/fisher_et_al_fig_1.png"></a>
</figure>

The study has an extensive discussion on factors that affect the growth of *Arcobacter* in sewage, and temperature, as usual, is one key player:

<figure>
	<a href="{{ site.url }}/images/oligotyping/fisher_et_al_fig_3.png"><img src="{{ site.url }}/images/oligotyping/fisher_et_al_fig_3.png"></a>
</figure>

You can read the study here: [doi: 10.3389/fmicb.2014.00525](http://journal.frontiersin.org/Journal/10.3389/fmicb.2014.00525/full)

## Dynamics of tongue microbial communities

<figure>
	<a href="http://journal.frontiersin.org/Journal/10.3389/fmicb.2014.00568/full"><img src="{{ site.url }}/images/oligotyping/mark_welch_et_al.png"></a>
</figure>

In this study, which I was a part of, Mark Welch *et al*. re-analyzes the infamous microbiome time series data published by [Caporaso *et al*.](http://www.ncbi.nlm.nih.gov/pubmed/21624126). We had recently [re-analyzed and published](http://www.pnas.org/content/111/28/E2875.full) the oral samples collected by the HMP project. The HMP data were mostly representing a cross-sectional sampling of a large group of individuals. The lack of multiple samples from each individual leaves many questions unanswered about the stability of the oral microbiome at the oligotypes level. For instance one of the interesting findings in our previous study was this (from Fig 4 in Eren *et al*.):

<figure>
	<a href="{{ site.url }}/images/oligotyping/eren_et_al_neisseria.png"><img src="{{ site.url }}/images/oligotyping/eren_et_al_neisseria.png"></a>
</figure>

In this figure each bar represents one individual, and colors show the distribution of *Neisseria* oligotypes in that person's tongue. What is interesting about this figure is that each individual is dominated by one of the *Neisseria* oligotypes (almost all of which are 99% identical to each other at the sequenced region and binned together with conventional methods). So, what is going on here? Are they functionally identical? Do they represent different stable community states? Are some of those bars coming from transient states? If we had looked at one of those individuals for a long period of time, would we see a sudden switch in the *Neisseria* population? And most importantly, what governs these patterns? Neutral effects? The host immune system? Rest of the microbiome? All of the above? Or none?

The dataset published in Caporaso *et al*.'s study, which contains 396 time points from only two individuals of course was a natural follow-up to our previous analysis of the HMP in an attempt to answer some of these questions. So. Jessica Mark Welch and Daniel Utter did the analysis, and here is the distribution of *Neisseria* oligotypes identified in these two individuals:

{: .notice}
I wish colors were identical to the previous study, but sequencing different regions of the 16S rRNA gene makes it very very hard to connect things to each other confidently

<figure>
	<a href="{{ site.url }}/images/oligotyping/mark_welch_et_al_fig_2.png"><img src="{{ site.url }}/images/oligotyping/mark_welch_et_al_fig_2.png"></a>
</figure>

There are a number of interesting things going on. At the very beginning of the sampling period, each individual is composed of one the Neisseria oligotype shown in green. Then they diverge into their own type, blue for female and cyan for male, and they stay like that! As a note, blue and cyan are one nucleotide apart from each other. Here is a note from the paper:

>These dynamics display two main characteristics which, taken together, may be termed a phase transition. The major behavior is one of stability. For most of the time, the oligotype distribution within an individual was essentially invariant, irrespective of whether the dominant oligotype in the individual was [Cyan] or [Blue]. The second property was of abrupt transition to an alternate oligotype. The time series data showed several instances in which a community initially dominated by one oligotype became transiently mixed and then transitioned to a state where one oligotype was dominant. These properties suggest that the evenly mixed populations of Neisseria on the tongue found in some individuals in the HMP data [shown in the previous figure] are transient states. Occasional replacement of the dominant oligotype argues against strong founder effects and priority effects for this taxon in the tongue microbiota. Throughout these transitions the fourth oligotype, [Purple], did not participate in the apparently competitive or exclusionary dynamics of types [Cyan] and [Blue], but persisted in relatively stable proportion in the community, likely demonstrating a subdivision of functional/ecological roles even among these very closely related taxa.

You can read the study here: [doi: 10.3389/fmicb.2014.00568](http://journal.frontiersin.org/Journal/10.3389/fmicb.2014.00568/full)


## An R package for the entropy decomposition

<figure>
	<a href="http://journal.frontiersin.org/Journal/10.3389/fmicb.2014.00601/full"><img src="{{ site.url }}/images/oligotyping/ramette_et_al.png"></a>
</figure>

One of the great surprises of this collection was to see Alban Ramette and Pier Luigi Buttigieg's R implementation of oligotyping and [Minimum Entropy Decomposition]({% post_url med/2014-11-04-med %}). The GitHub repository for 'otu2ot' is located here: [https://github.com/aramette/otu2ot](https://github.com/aramette/otu2ot).

Ramette *et al*.'s R library makes it much easier for R users to start using the approach on their datasets. The R library not only almost completely matches the functionality of a full oligotyping installation, but it also comes with two novel features: Broken stick model, to identify which oligotypes are more abundant than expected by chance, and a one-pass procedure, to rapidly assess the amount of microdiversity present in a group of sequences after only one round of entropy calculation. 

The oligotyping pipeline comes all sorts of bells and whistles in an attempt to improve the user experience. The R implementation, on the other hand, is more likely to be used by statisticians and developers to test and improve the approach. Broken stick model is a great example to that, and opens up a great path to better and statistically sound noise filtering on this type of data.

You can read the study here: [doi: 10.3389/fmicb.2014.00601](http://journal.frontiersin.org/Journal/10.3389/fmicb.2014.00601/full)

## And others

There are four other publications in the collection, one of which is still in press as of today:

- [**Oligotyping reveals community level habitat selection within the genus Vibrio**](http://journal.frontiersin.org/Journal/10.3389/fmicb.2014.00563)<br>
Victor T. Schmidt, Julie Reveillaud, Erik Zettler, Tracy J. Mincer, Leslie Murphy and Linda A. Amaral-Zettler.

- "[**Phaeocystis antarctica blooms strongly influence bacterial community structures in the Amundsen Sea polynya**](http://journal.frontiersin.org/Journal/10.3389/fmicb.2014.00646)"<br>Tom O. Delmont, Katherine M. Hammar, Hugh W. Ducklow, Patricia L. Yager and Anton F. Post.

- "[**Biogeographic patterns of bacterial microdiversity in Arctic deep-sea sediments (HAUSGARTEN, Fram Strait)**](http://journal.frontiersin.org/Journal/10.3389/fmicb.2014.00660)"<br>Pier Luigi Buttigieg and Alban Ramette.

- "[**Oligotyping reveals stronger relationship of organic soil bacterial community structure with N-amendments and soil chemistry in comparison to that of mineral soil at Harvard Forest, MA, USA**](http://journal.frontiersin.org/Journal/10.3389/fmicb.2015.00049)"<br>Swathi Anuradha Turlapati, Rakesh Minocha, Stephanie Long, Jordan Ramsdell and Subhash C Minocha.

## Final Words

The diversity of environments studied in these publications, and the rate of the recovery of ecologically meaningful findings show that there is a great potential for highly resolved depictions of microbiomes. I am very thankful for everyone who contributed to this collection.

[Oligotyping]({{ site.url }}/projects/oligotyping/) and [minimum entropy decomposition]({{ site.url }}/projects/med/) provided a framework for researchers to explore these dimensions of their datasets at levels of single nucleotide resolution. I am sure there will be many approaches going forward, and a lot of research to make results accurate.

It is just getting the mindset of microbial ecology out of the local minimum called "3%". The rest will follow very quickly. 

---

This has been a great experience on many levels, and I am very thankful to [Frontiers in Systems Microbiology](http://www.frontiersin.org/systems_microbiology) and the great team behind it for this opportunity.

I think research topics in Frontiers provide a great framework to create a coherent collection of work that aims to contribute to a particular and well-defined issue.

If you think you have a study that would fit well into this collection, please send me or `microbiology.researchtopics at frontiersin dot org` an e-mail.






