---
layout: page
title: About
tags: [meren]
modified: 2015-02-02T20:53:07.573882-04:00
comments: false
image:
  feature: eel-pond.jpg
  credit: Meren
  creditlink: http://meren.org
---

{% include _toc.html %}

[http://meren.org](http://meren.org)

## Contact

<div style="width: 100%; background: #F7F7FA; padding: 5px; padding-top: 8px; margin: 5px; margin-bottom: 10px;"><strong>E-mail</strong>: meren@mbl.edu / <strong>Address</strong>: Marine Biological Laboratory, JBPC (L-326), 7 MBL Street, Woods Hole, MA 02543 USA / <strong>Office Phone</strong>: +1-508-286-7112 / <strong>Fax</strong>: +1-508-457-4727 / <strong>Skype</strong>: a.murat.eren</div>

## Selected Publications

- **Eren AM**, et al. (2014). "[Minimum entropy decomposition: Unsupervised oligotyping for sensitive partitioning of high-throughput marker gene sequences](http://onlinelibrary.wiley.com/doi/10.1111/2041-210X.12114/full)". *The ISME Journal*.

- **Eren AM**, et al. (2014). "[Oligotyping analysis of the human oral microbiome](http://www.pnas.org/cgi/doi/10.1073/pnas.1409644111)". *PNAS*.

- **Eren AM**, et al. (2013). "[Oligotyping: differentiating between closely related microbial taxa using 16S rRNA gene data](http://onlinelibrary.wiley.com/doi/10.1111/2041-210X.12114/full)". *Methods in Ecology and Evolution*.

- **Eren AM**, et al. (2014). "[A single genus in the gut microbiome reflects host preference and specificity](http://www.nature.com/ismej/journal/vaop/ncurrent/full/ismej201497a.html)". *The ISME Journal*.

- **Eren AM**, et al. (2011). "[Exploring the diversity of Gardnerella vaginalis through subtle nucleotide variation](http://www.plosone.org/article/info%3Adoi%2F10.1371%2Fjournal.pone.0026732)". *PLoS ONE*.

- **Eren AM**, et al. (2013). "[A Filtering Method to Generate High Quality Short Reads Using Illumina Paired-End Technology](http://www.plosone.org/article/info%3Adoi%2F10.1371%2Fjournal.pone.0066643)". *PLoS ONE*.

- **Eren AM**, et al. (2013). "[DRISEE overestimates errors in metagenomic sequencing data](http://bib.oxfordjournals.org/content/early/2013/05/22/bib.bbt010.abstract)". *Briefings in Bioinformatics*.

- McLellan SL and **Eren AM** (2014). "[Discovering new indicators of fecal pollution](http://www.sciencedirect.com/science/article/pii/S0966842X14001589)". *Trends in Microbiology*.

A full list can be seen in my [Mendeley profile](http://www.mendeley.com/profiles/a-murat-eren/).

---

## Reserach Interests

**Human microbiome**. Diverse assemblage of bacteria that harbor all surfaces of the human body makes us great environments to study ecological questions, and to develop a better understanding of the function of bacterial communities from a public health perspective. I study human vaginal, gastrointestinal and oral communities to find associations between microbial community structures and health and disease states of their hosts.

**Marker gene anlayses**. Marker genes amplified from environmental samples are indispensable tools for deep and detailed characterization of microbial population structures, including very rare members they harbor. The 16S rRNA gene has been historically a very useful gene to target and over the years we have generated tremendous amount of data. I develop tools that allow researchers to partition their high-throughput sequencing datasets into highly resolved and ecologically relevant units.

**Metagenomics**. Although marker genes are indispensable for the characterization of the community structure, they cannot answer questions about these communities regarding what they are capable of doing with respect to their total genomic content with confidence. Although there are challenges, metagenomic approaches that utilize shotgun sequencing of environmental DNA is key to understand the overall functional potential of microbial communities and I am interested in working with metagenomics datasets to look for answers marker gene analyses cannot provide.

**Quality Filtering**. Adequate quality filtering of raw sequences is a critical aspect of all studies that rely on next-generation high-througput sequencing data. I develop and maintain tools and methods to perform quality filtering on raw reads generated by massively-parallel sequencing platforms.

---

## Education


<table>
<tbody>
<tr>
<td width="15%">2011-2013</td>
<td>Post-Doctoral Training, Marine Biological Laboratory
Supervised by Dr. <a href="http://www.mbl.edu/jbpc/faculty/mitchell-sogin/">Mitchell L. Sogin</a>.</td>
</tr>
<tr>
<td background="gray">2007-2011</td>
<td>Ph.D., University of New Orleans
Supervised by <a href="http://www.medschool.lsuhsc.edu/microbiology/faculty_detail.aspx?name=ferris_michael">Michael J. Ferris</a> (Microbial Ecology), <a href="http://www.medschool.lsuhsc.edu/microbiology/faculty_detail.aspx?name=taylor_christopher">Christopher M. Taylor</a> (Computer Science).</td>
</tr>
<tr>
<td>1998-2002</td>
<td>B.S., Canakkale Onsekiz Mart Universitesi, Turkey.</td>
</tr>
</tbody>
</table>
